latex/epoch.tex

```diff
@@ -26,36 +31,171 @@
+    covering the reward calculation which happens between (E) and (F).
+\end{itemize}
+
+\subsection{Helper Functions}
+\label{sec:stake-dist}
+
+Figure~\ref{fig:funcs:epoch-helper} defines three helper functions needed
+throughout the rest of the section.
+
+\begin{itemize}
+  \item The function $\fun{obligation}$ calculates the the minimal amount of coin needed to
+    pay out all depot refunds, as of the current slot.
+  \item The function $\fun{poolRefunds}$ is used to calculate the total refunds
+    that must be distributed for stake pools scheduled to retire.
+    Note that this calculation takes a slot number corresponding to the epoch boundary slot
+    when the calculation is performed.  The map returned maps pool operator hashkeys to the
+    refunds, which will ultimately be returned to the registered reward account.
+  \item The function $\fun{updateAvgs}$ calculates the new performance moving averages.
+\end{itemize}
+
+
+%%
+%% Figure - Helper Functions for Epoch Rules
+%%
+\begin{figure}[htb]
+  \emph{Total possible refunds}
+  \begin{align*}
+      & \fun{obligation} \in \PParams \to \StakeKeys \to \StakePools \to \Slot \to \Coin \\
+      & \obligation{pp}{stkeys}{stpools}{cslot} =\\
+      & \sum\limits_{(\_ \mapsto s) \in \var{stkeys}}
+        \refund{d_{\mathsf{val}}}{d_{\min}}{\lambda_d}{(\slotminus{cslot}{s})}
+        + \sum\limits_{(\_ \mapsto s) \in \var{stpools}}
+        \refund{p_{\mathsf{val}}}{p_{\min}}{\lambda_p}{(\slotminus{cslot}{s})} \\
+      &
+      \begin{array}{lr@{~=~}l}
+        \where
+          & \dval,~d_{\min},~\lambda_d
+          & \fun{keyDeposit}~\var{pp},~\fun{keyMinRefund}~\var{pp},~\fun{keyDecayRate}~\var{pp}
+          \\
+          & p_{\mathsf{val}},~p_{\min},~\lambda_p
+          & \fun{poolDeposit}~\var{pp},~\fun{poolMinRefund}~\var{pp},~\fun{poolDecayRate}~\var{pp}
+      \end{array}\\
+  \end{align*}
+  \emph{Pool refunds}
+  \begin{align*}
+      & \fun{poolRefunds} \in \PParams \to (\HashKey_{pool} \mapsto \Epoch) \to \Slot \to
+      (\HashKey_{pool} \mapsto \Coin) \\
+      & \poolRefunds{pp}{retiring}{cslot} = \left\{
+        \var{hk}\mapsto
+          \refund{p_{\mathsf{val}}}{p_{\min}}{\lambda}{(\slotminus{cslot}{(\fun{slot}~e)})}
+          \mid
+          \var{hk}\mapsto e\in\var{retiring}
+        \right\}\\
+      & \where p_{\mathsf{val}},~p_{\min},~\lambda_p =
+          \fun{poolDeposit}~\var{pp},~\fun{poolMinRefund}~\var{pp},~\fun{poolDecayRate}~\var{pp} \\
+  \end{align*}
+
+  \emph{Update Moving Averages}
+  \begin{align*}
+      & \fun{updateAvgs} \in \PParams \to \Avgs \to \BlocksMade \to
+          (\HashKey_{pool}\mapsto \Stake) \to \Avgs \\
+      & \updateAvgs{pp}{avgs}{blocks}{pooledStake} = \\
+      & ~~~ \left\{hk\mapsto \movingAvg{pp}{hk}{n}{\overline{N}}{avgs}
+            \mid
+            hk\mapsto n\in\var{blocks}, hk\mapsto\overline{N}\in\var{expectations}
+            \right\} \\
```

Comment:
I've done this is a few spots, where the comprehension treats membership in a second set as a predicate. Maybe it's confusing, though.  I like the idea of making expectations a function! I hadn't yet defined any functions in a where clause, but I think it would help here.

Reply:
Ah, I see. That makes sense.